python/ray/serve/tests/test_grpc.py

```diff
@@ -603,54 +612,45 @@ def test_grpc_proxy_timeouts(ray_instance):
         ),
     )
 
-    @serve.deployment()
+    signal_actor = SignalActor.remote()
+
+    @serve.deployment
     class HelloModel:
         def __call__(self, user_message):
-            time.sleep(5)
+            ray.get(signal_actor.wait.remote())
             return serve_pb2.UserDefinedResponse(greeting="hello")
 
         def Streaming(self, user_message):
             for i in range(10):
-                time.sleep(5)
+                ray.get(signal_actor.wait.remote())
                 yield serve_pb2.UserDefinedResponse(greeting="hello")
 
-    model = HelloModel.bind()
-    app_name = "app1"
-    replica_name = DeploymentID("HelloModel", app_name)
-    serve.run(target=model, name=app_name)
-    replicas = ray.get(
-        serve.context._global_client._controller._all_running_replicas.remote()
-    )
```

Comment:
@GeneDer note that this part is unnecessary (because `serve.run` blocks until the app is healthy) and it was also using private APIs which we should avoid where possible. Otherwise now if we change `_all_running_replicas` or `DeploymentID` (both implementation details), we'd need to update this test.

Reply:
Gotcha!